### DISCUSSION TITLE
How to display action response in a modal

### DISCUSSION THREAD

**Discussion by hrisitjansiwb** [filamentphp/filament#6007] [Help]
I have an action that needs to ping a server and return result back if the server is available or not.

```
Actions\Action::make('ping')
                ->action(function(array $data) {
                   $response = Http::withToken($this->record->endpoint_password)
                       ->get($this->record->endpoint_url);
                   $this->record->responseStatus = $response->status();

                   return $response->status();
                })
```

I wonder how can I display that result in a modal view? 

If i call **modalContent** then the modal is displayed immediately and I don't have access to the response result. 

In the **after**  method I can access to **$this->record->responseStatus** I get the desired result, but if there is a way to display the modal from the **after** method?

**Comment by danharrin** [comment]
Maybe use a notification?

**Reply by hrisitjansiwb** [reply]
Modal is more suitable for this case.

But how I can display notification on action complete?

**Reply by danharrin** [reply]
https://filamentphp.com/docs/2.x/notifications/sending-notifications
